schemaRegistryClient.go

```diff
@@ -394,6 +396,73 @@ func (client *SchemaRegistryClient) CreateSchema(subject string, schema string,
 	return newSchema, nil
 }
 
+// LookupSchema looks up the schema by subject and schema string. If it finds the schema it returns it with all its associated information.
+func (client *SchemaRegistryClient) LookupSchema(subject string, schema string, schemaType SchemaType, references ...Reference) (*Schema, error) {
+	switch schemaType {
+	case Avro, Json:
+		compiledRegex := regexp.MustCompile(`\r?\n`)
+		schema = compiledRegex.ReplaceAllString(schema, " ")
+	case Protobuf:
+		break
+	default:
+		return nil, fmt.Errorf("invalid schema type. valid values are Avro, Json, or Protobuf")
+	}
+
+	if references == nil {
+		references = make([]Reference, 0)
+	}
+
+	schemaReq := schemaRequest{Schema: schema, SchemaType: schemaType.String(), References: references}
+	schemaBytes, err := json.Marshal(schemaReq)
+	if err != nil {
+		return nil, err
+	}
+	payload := bytes.NewBuffer(schemaBytes)
+	resp, err := client.httpRequest("POST", fmt.Sprintf(subjectBySubject, subject), payload)
+	if err != nil {
+		return nil, err
+	}
+
+	schemaResp := new(schemaResponse)
+	err = json.Unmarshal(resp, &schemaResp)
+	if err != nil {
+		return nil, err
+	}
+
+	var codec *goavro.Codec
+	if client.getCodecCreationEnabled() {
+		codec, err = goavro.NewCodec(schemaResp.Schema)
```

Comment:
Could you add a schemaType checker before creatng an AvroCodec? Small chance that codec creation is set to true, but a protobuf or json schema is fetched.

Reply:
done